component/src/main/java/org/wso2/extension/siddhi/execution/geo/GeoCoordinateStreamFunctionProcessor.java

```diff
@@ -0,0 +1,216 @@
+     *
+     * @param data null if the function parameter count is zero or runtime data value of the function parameter
+     * @return the data for additional output attribute introduced by the function
+     */
+    @Override
+    protected Object[] process(Object data) {
+        String ip = data.toString();
+        GeoCoordinate geoCoordinate = geoCoordinateResolverImpl.getGeoCoordinateInfo(ip);
+        return new Object[]{geoCoordinate.getLatitude(), geoCoordinate.getLongitude()};
+    }
+
+    /**
+     * The initialization method for {@link StreamFunctionProcessor},
+     * which will be called before other methods and validate
+     * the all configuration and getting the initial values.
+     *
+     * @param inputDefinition              the incoming stream definition
+     * @param attributeExpressionExecutors the executors of each function parameters
+     * @param configReader                 this hold the {@link StreamFunctionProcessor} extensions
+     *                                     configuration reader.
+     * @param siddhiAppContext             the context of the siddhi app
+     * @return the output's additional attributes list introduced by the function
+     */
+    @Override
+    protected List<Attribute> init(AbstractDefinition inputDefinition,
+                                   ExpressionExecutor[] attributeExpressionExecutors, ConfigReader configReader,
+                                   SiddhiAppContext siddhiAppContext) {
+        if (!isExtensionConfigInitialized.get()) {
+            initializeExtensionConfigs(configReader);
+        }
+        if (attributeExpressionExecutors[0].getReturnType() != Attribute.Type.STRING) {
+            throw new SiddhiAppCreationException("First parameter should be of type string");
+        }
+        List<Attribute> attributes = new ArrayList<Attribute>(2);
+        attributes.add(new Attribute("latitude", Attribute.Type.DOUBLE));
+        attributes.add(new Attribute("longitude", Attribute.Type.DOUBLE));
+        return attributes;
+    }
+
+    /**
+     * This will be called only once and this can be used to acquire
+     * required resources for the processing element.
+     * This will be called after initializing the system and before
+     * starting to process the events.
+     */
+    @Override
+    public void start() {
+
+    }
+
+    /**
+     * This will be called only once and this can be used to release
+     * the acquired resources for processing.
+     * This will be called before shutting down the system.
+     */
+    @Override
+    public void stop() {
+
+    }
+
+    /**
+     * Used to collect the serializable state of the processing element, that need to be
+     * persisted for reconstructing the element to the same state on a different point of time
+     *
+     * @return stateful objects of the processing element as an map
+     */
+    @Override
+    public Map<String, Object> currentState() {
+        return Collections.emptyMap();
```

Comment:
Previously it was like "return new HashMap<String, Object>();". 
In the review, it suggested changing it as "Collections.emptyMap()" since we are not using this.